tests/unit/background.spec.js

```diff
@@ -1,39 +1,192 @@
-const { installedListener } = require("../../src/background");
+const {
+  installedListener,
+  onClickedListener,
+} = require("../../src/background");
 
+const createTabs = jest.spyOn(chrome.tabs, "create");
 const sanitizeSettings = jest.spyOn(ConfigFile.default, "sanitizeSettings");
 const updateNow = jest.spyOn(ConfigFile.default, "updateNow");
 
 describe("background.js", () => {
-  it("should open migration.html if previous version was 4", () => {
+  it("Should open migration.html if previous version was 4", () => {
     installedListener({ previousVersion: "4" });
-    expect(chrome.tabs.create.calledOnce).toBe(true);
-    expect(
-      chrome.tabs.create.withArgs({
-        url: "migration.html?previous=4",
-        selected: true,
-      }).calledOnce
-    ).toBe(true);
+    expect(createTabs).toHaveBeenCalled();
+    expect(createTabs).toBeCalledWith({
+      url: "migration.html?previous=4",
+      selected: true,
+    });
   });
 
-  it("should call sanitizeSettings if previous version was not 4", () => {
+  it("Should call sanitizeSettings if previous version was not 4", () => {
     installedListener({ previousVersion: "5" });
-    expect(
-      chrome.tabs.create.withArgs({
-        url: MiscURLs.INSTALLED_URL,
-        selected: true,
-      }).calledOnce
-    ).toBe(true);
+    expect(createTabs).toBeCalledWith({
+      url: MiscURLs.INSTALLED_URL,
+      selected: true,
+    });
     expect(sanitizeSettings).toHaveBeenCalled();
     expect(updateNow).not.toHaveBeenCalled();
   });
 
-  it("should call updateNow if installing for the first time", async () => {
+  it("Should call updateNow if installing for the first time", async () => {
     await installedListener({ reason: "install" });
     expect(sanitizeSettings).toHaveReturned();
     expect(updateNow).toHaveBeenCalled();
   });
+
+  it("Should open tab for simple search provider", async () => {
+    const info = {
+      menuItemId: "searchprovider-0",
+      selectionText: "test",
+    };
+    const tab = {
+      index: 10,
+    };
+    onClickedListener(info, tab);
+
+    const providers = await LocalStore.getOne(StoreKey.SEARCH_PROVIDERS);
+    const settings = await LocalStore.getOne(StoreKey.SETTINGS);
+    const provider = _.find(providers, function (item) {
+      return item.menuIndex === info.menuItemId;
+    });
+    let targetURL = getProviderTargetURL(provider, info.selectionText);
+    expect(createTabs).toBeCalledWith({
+      url: targetURL,
+      selected: !settings.resultsInBackgroundTab,
+      index: tab.index + 1,
+    });
+
+    // Test providers with a POST request.
+    providers[0].postEnabled = true;
+    await LocalStore.setOne(StoreKey.SEARCH_PROVIDERS, providers);
+    await onClickedListener(info, tab);
+    targetURL = getProviderTargetURL(provider, info.selectionText);
+    expect(createTabs).toBeCalledWith({
+      url: targetURL,
+      selected: !settings.resultsInBackgroundTab,
+      index: tab.index + 1,
+    });
```

Comment:
The test for the POST request should be on a separate `it(` (like `it("Should open dialog for POST search provider", async () => {`).

Also I don't see the unit tests for a search provider with a Proxy